### DISCUSSION TITLE
Can I use PostgreSQL?

### DISCUSSION THREAD

**Discussion by tomdavidson** [bigcapitalhq/bigcapital#259] [Q&A]
Hello, I see knex in the code base. Can we use PostreSQL or SQLite even?

**Comment by TayfurYldz** [comment]
Knex itself supports multiple database engines, but Bigcapital currently does not expose the database client as a deployment choice.

I checked current `main` (`4353a16b545de02d4882c30656ca34bad1bbbfdf`). Both database configurations are explicitly pinned to MySQL:

```ts
// packages/server/src/common/config/system-database.ts
client: 'mysql'

// packages/server/src/common/config/tenant-database.ts
client: 'mysql'
```

The server package also ships the MySQL drivers, and the development/production compose files provision a MySQL service. So changing only Knex configuration or pointing `DB_HOST` at PostgreSQL/SQLite is not a supported substitution.

There is another architectural detail: Bigcapital has both a system database and per-tenant database configuration/migrations. Supporting a different engine therefore means more than adding a Knex driver; all migrations/schema operations and any MySQL-specific queries need to be made portable and tested for both database layers.

So for an unmodified/self-hosted Bigcapital deployment, use MySQL (or the MySQL-compatible setup supplied by the project). PostgreSQL/SQLite would currently require an application-level port rather than a configuration switch.
